src/config/index.ts

```diff
@@ -0,0 +1,46 @@
+import { Octokit } from "probot";
+import { loadConfigFromGithub } from "./loader";
+import ConfigSchema, { Config, RulesConfig } from "./schema";
+import presets, { defaultPreset } from "../lint/presets";
+
+const configFileName = ".gitlintrc";
+const extensions = ["", ".yaml", ".yml"];
+
+export async function getConfigFromGithub(
```

Comment:
This function has three responsibilities:
- discovering possible configuration by possible extensions
- loading config from a remote source (GitHub)
- parse and process the config file (e.g. apply preset)

Ideally, these three responsibilities should be refactored out as separate functions.

Also, conceptually there are two modules: linter and bot. Configuration should belongs to the linter, and it should not be aware of GitHub. Instead, the bot module would have a configuration source, which linter can use to load config.